server/src/main/java/org/opensearch/index/SegmentReplicationStatsTracker.java

```diff
@@ -28,11 +29,24 @@ public class SegmentReplicationStatsTracker {
     private final IndicesService indicesService;
     private final Map<ShardId, AtomicInteger> rejectionCount;
 
+    @Inject
     public SegmentReplicationStatsTracker(IndicesService indicesService) {
         this.indicesService = indicesService;
         rejectionCount = ConcurrentCollections.newConcurrentMap();
     }
 
+    public SegmentReplicationRejectionStats getRejectionStats() {
+        long rejectionCount = 0;
+        for (IndexService indexService : indicesService) {
+            for (IndexShard indexShard : indexService) {
```

Comment:
lets read the isSegRepEnabled setting off indexService before iterating all shards.

Reply:
Added in latest commit